src/app/components/common/courthouse/courthouse.component.ts

```diff
@@ -12,43 +10,34 @@ import { HttpErrorResponse } from '@angular/common/http';
   templateUrl: './courthouse.component.html',
   styleUrls: ['./courthouse.component.scss'],
 })
-export class CourthouseComponent implements OnInit {
-  @ViewChild('courthouseAutocomplete', { static: false }) autocomplete!: ElementRef<HTMLElement>;
-  loaded = false;
-  loadError = false;
-  courthouses: CourthouseData[] = [];
+export class CourthouseComponent implements AfterViewInit {
+  @ViewChild('courthouseAutocomplete') autocompleteContainer!: ElementRef<HTMLElement>;
+
+  @Input() courthouses: CourthouseData[] = [];
```

Comment:
The idea behind the courthouses component is to handle all the loading / displaying of courthouses in separation from the parent. These changes mean anytime we want to use this the parent component is responsible for loading the courthouses.

Could this go back, or it is a change that's required to get the error handling working?

Reply:
I moved it out because it's supposed to be a dumb component just responsible for displaying the data and not fetching the data. Typically in angular the parent fetches all the necessary data for the screen and passes it to all UI/Child/dumb components. It's the container / presentation pattern. Calling services from a child is considered an anti-pattern. It's still reusable and it isn't much work to async pipe and pass the data in.

```
<app-courthouse-field
          *ngIf="courthouses$ | async as courthouses"
          [courthouses]="courthouses"
          (courthouseSelect)="onCourthouseSelected($event)"
          [isInvalid]="isControlInvalid('courthouse')">
</app-courthouse-field>
```

Reply:
Ok, that's cool, thanks for explaining.